ext/phar/phar_object.c

```diff
@@ -2306,6 +2306,13 @@ static zend_object *phar_convert_to_other(phar_archive_data *source, int convert
 			/* exception already thrown */
 			return NULL;
 		}
+
+		if (newentry.fp == NULL) {
```

Comment:
This looks like the wrong solution. You're resetting the file pointer but `phar_copy_file_contents` already used it.
I believe the right solution is to:
- Reset fp and cfp right after the copy from entry to newentry. The new file pointer will then be opened and set correctly in `phar_copy_file_contents`
- Remove the `/* save for potential restore on error */` block that moves the file pointers as this recovery mechanism isn't implemented as far as I see.

Reply:
It's hard for me to understand why phar codes use `cfp`, so I use this strange solution to avoid break other features.

Also, there are many memcpys (like newentry <- entry here), dont know why, so I chose to not modify `phar_copy_file_contents`